src/main/scala/stuff/functors.scala

```diff
@@ -68,13 +78,24 @@ object Functor {
 
     type F[Z <: F0#Source#Objects] = G0#F[ F0#F[Z] ]
 
-    def at[X <: Source#Objects, Y <: Source#Objects]: Source#C[X,Y] -> Target#C[F[X], F[Y]] =
+    def at[
+      X <: Source#Objects,
+      Y <: Source#Objects
+    ]
+    : Source#C[X,Y] -> Target#C[F[X], F[Y]] =
       first.at >-> second.at
   }
 
-  def composition[F0 <: Functor, G0 <: Functor { type Source = F0#Target }]: (is[F0] × is[G0]) -> Composition[F0,G0] =
-    λ { fg => new Composition(left(fg), right(fg)) }
+  def composition[
+    F0 <: Functor,
+    G0 <: Functor { type Source = F0#Target }
+  ]
+  : (is[F0] × is[G0]) -> is[Composition[is[F0],is[G0]]] =
+    λ { fg =>
+      new Composition(left(fg), right(fg)).asInstanceOf[is[Composition[is[F0],is[G0]]]]
```

Comment:
What does this mean? Why do we need the `.asInstanceOf[is[Composition[is[F0],is[G0]]]]` part?

Reply:
See https://github.com/ohnosequences/stuff/pull/52#issuecomment-336859795. We shouldn't need it.